src/realm/db.cpp

```diff
@@ -909,7 +909,7 @@ void DB::open(const std::string& path, bool no_create_file, const DBOptions& opt
     REALM_ASSERT(path.size());
 
     m_db_path = path;
-    m_path_hash = StringData(path).hash() & 0xffff;
+    m_path_hash = (size_t(this) >> 4) & 0xffff;
```

Comment:
What is this line doing? Is really not clear...

Reply:
Just hashing the file name was not good enough. If two DBs were opening the same file, they would get the same hash. Here we just  take some bytes out of the object pointer. That should be fine.

Reply:
make that into a comment :-)

Reply:
OK, maybe I am wrong, but it seems that this is basically an important part for fixing this issue. And it is really really difficult to make sense of this line just reading it, without any added context. 
Can you please add a comment or move this logic for hashing the db object into some helper method or some class.

Reply:
`m_path_hash` is only used for the prefix of the logger. It was changed for debugging purposes so that the log output can be differentiated between DB instances. In that way it is not critical to the fix, but it helped find out what was happening. The essence of the fix is separation of GroupWriter and GroupCommitter, so that async commits can just make a commit without doing any of the online compaction.